lib/rdoc/generator/darkfish.rb

```diff
@@ -780,4 +780,19 @@ def template_for file, page = true, klass = ERB
     template
   end
 
+  # Returns an excerpt of the content for usage in meta description tags
+  def excerpt(content)
+    text = content.is_a?(RDoc::Comment) ? content.text : content
+
+    # Match from a capital letter to the first period, discarding any links, so
+    # that we don't end up matching badges in the README
+    first_paragraph_match = text.match(/[A-Z][^\.:\/]+\./)
```

Comment:
Not All paragraphs start with a capital letter, as usual English text.
E.g., README.md in drb has the paragraph:

> dRuby is a distributed object system for Ruby.  It allows an object in one
> Ruby process to invoke methods on an object in another Ruby process on the
> same or a different machine.

I think it should be any text, from the alphabet to the first period, excluding some symbols.
However, this is still too strict for most natural languages.